src/ray/object_manager/object_buffer_pool.cc

```diff
@@ -327,8 +327,31 @@ ray::Status ObjectBufferPool::EnsureBufferExists(const ObjectID &object_id,
 }
 
 void ObjectBufferPool::FreeObjects(const std::vector<ObjectID> &object_ids) {
-  absl::MutexLock lock(&pool_mutex_);
-  RAY_CHECK_OK(store_client_->Delete(object_ids));
+  constexpr int kMaxAttempts = 3;
+  absl::Duration backoff = absl::Milliseconds(5);
+  int attempt = 1;
+  while (attempt <= kMaxAttempts) {
+    Status s;
+    {
+      absl::MutexLock lock(&pool_mutex_);
+      s = store_client_->Delete(object_ids);
+    }
+    if (s.ok()) {
+      return;
+    }
+    if (!s.IsIOError()) {
+      RAY_LOG(WARNING) << "Plasma delete failed (non-IOError), not retrying: " << s;
+      return;
+    }
```

Comment:
Should we make this fatal and only keep IOError non-fatal and retryable?

Reply:
potentially from `PlasmaClient::Delete` code path, `Status::Invalid(...)` if delete request construction fails.